src/Persistence/Array_.php

```diff
@@ -119,11 +119,22 @@ public function tryLoad(Model $model, $id, string $table = null): ?array
         return $this->typecastLoadRow($model, $this->data[$table][$id]);
     }
 
+    /**
+     * Tries to load first available record and return data record.
+     */
+    public function loadAny(Model $model, string $table = null): ?array
+    {
+        $row = $this->tryLoadAny($model, $table);
+        if (!$row) {
+            throw new Exception('No matching records were found!', 404);
```

Comment:
does this exception match the SQL persistence incl. the code?

Reply:
closely, the sql exeption adds more infos:
```

            throw (new Exception('No matching records were found', 404))
                ->addMoreInfo('model', $model)
                ->addMoreInfo('scope', $model->scope()->toWords());
        }
```

To me it seems there is quite some duplicate code between Array_ and SQL Persistences. Maybe we can sort this in #690?

Reply:
with https://github.com/atk4/data/issues/699 we can drop Array completely...